test/fuzz/transformation_make_vector_operation_dynamic_test.cpp

```diff
@@ -0,0 +1,328 @@
+         %36 = OpCompositeExtract %6 %21 1
+         %37 = OpCompositeInsert %6 %15 %17 0
+         %38 = OpCompositeInsert %6 %16 %17 1
+         %39 = OpCompositeInsert %6 %13 %18 0
+         %40 = OpCompositeInsert %6 %14 %18 1
+         %41 = OpCompositeInsert %7 %13 %19 0
+         %42 = OpCompositeInsert %7 %14 %19 1
+         %43 = OpCompositeInsert %7 %15 %19 2
+         %44 = OpCompositeInsert %8 %13 %20 0
+         %45 = OpCompositeInsert %8 %14 %20 1
+         %46 = OpCompositeInsert %8 %15 %20 2
+         %47 = OpCompositeInsert %8 %16 %20 3
+         %48 = OpCompositeInsert %9 %17 %21 0
+         %49 = OpCompositeInsert %9 %18 %21 1
+               OpReturn
+               OpFunctionEnd
+  )";
+
+  const auto env = SPV_ENV_UNIVERSAL_1_5;
+  const auto consumer = nullptr;
+  const auto context =
+      BuildModule(env, consumer, reference_shader, kFuzzAssembleOption);
+  ASSERT_TRUE(IsValid(env, context.get()));
+
+  FactManager fact_manager;
+  spvtools::ValidatorOptions validator_options;
+  TransformationContext transformation_context(&fact_manager,
+                                               validator_options);
+
+  // Tests undefined instruction.
+  auto transformation = TransformationMakeVectorOperationDynamic(50, 10);
+  ASSERT_FALSE(
+      transformation.IsApplicable(context.get(), transformation_context));
+
+  // Tests non-composite instruction.
+  transformation = TransformationMakeVectorOperationDynamic(23, 11);
+  ASSERT_FALSE(
+      transformation.IsApplicable(context.get(), transformation_context));
+
+  // Tests composite being a matrix.
+  transformation = TransformationMakeVectorOperationDynamic(48, 12);
+  ASSERT_FALSE(
+      transformation.IsApplicable(context.get(), transformation_context));
+
+  // Tests literal not defined as constant.
+  transformation = TransformationMakeVectorOperationDynamic(34, 51);
+  ASSERT_FALSE(
+      transformation.IsApplicable(context.get(), transformation_context));
+
+  // Tests applicable instructions.
+  transformation = TransformationMakeVectorOperationDynamic(24, 10);
+  ASSERT_TRUE(
+      transformation.IsApplicable(context.get(), transformation_context));
+
+  transformation = TransformationMakeVectorOperationDynamic(25, 11);
+  ASSERT_TRUE(
+      transformation.IsApplicable(context.get(), transformation_context));
+
+  transformation = TransformationMakeVectorOperationDynamic(26, 10);
+  ASSERT_TRUE(
+      transformation.IsApplicable(context.get(), transformation_context));
+
+  transformation = TransformationMakeVectorOperationDynamic(37, 10);
+  ASSERT_TRUE(
+      transformation.IsApplicable(context.get(), transformation_context));
+
+  transformation = TransformationMakeVectorOperationDynamic(38, 11);
+  ASSERT_TRUE(
+      transformation.IsApplicable(context.get(), transformation_context));
+
+  transformation = TransformationMakeVectorOperationDynamic(39, 10);
+  ASSERT_TRUE(
+      transformation.IsApplicable(context.get(), transformation_context));
+}
```

Comment:
Interesting. We normally apply the transformations and then compare the module to some SPIR-V assembly and check that it is valid. I feel like you might as well do this? But I can see that you do this kind of thing below in a separate test.

Reply:
It is in the `Apply` test.